llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -242,21 +243,15 @@ static void collectKernelModuleMap(
     KernelMapEntryScope EntryScope) {
 
   for (auto &F : M.functions()) {
-    if (F.getCallingConv() == CallingConv::SPIR_KERNEL) {
+    // Process module entry points: kernels and SYCL_EXTERNAL functions.
+    // Only they have sycl-module-id attribute, so any other unrefenced
```

Comment:
I think we also need a test which checks that unreferenced functions w/o external linkage (declared in namespace {} or as static) are really dropped, otherwise potential code bloat would be undetected.

Reply:
I thought we have such a test but looks like we don't. Will add.